Neos.Neos/Classes/Fusion/Cache/ContentCacheFlusher.php

```diff
@@ -86,13 +86,25 @@ public function registerNodeChange(NodeInterface $node)
     }
 
     /**
-     * Fetches possible usages of the asset and registers nodes that use the asset as changed.
+     * Deprecated. Please use ContentCacheFlush::registerAssetChange
      *
+     * @deprecated
      * @param AssetInterface $asset
      * @return void
      */
     public function registerAssetResourceChange(AssetInterface $asset)
     {
+        $this->registerAssetChange($asset);
+    }
+
+    /**
+     * Fetches possible usages of the asset and registers nodes that use the asset as changed.
+     *
+     * @param AssetInterface $asset
+     * @return void
+     */
+    public function registerAssetChange(AssetInterface $asset)
+    {
         if (!$asset->isInUse()) {
```

Comment:
Good point, I believe the signal was created before the `isInUse` method was added and it wasn't added to the interface to avoid being a breaking change. Now we have the opportunity to change it as we're introducing a new method. I think it would be best to use `Asset` instead for the new method to ensure it has the method `isInUse`. Alternatively we stick to the interface and check if the `isInUse` method exists to avoid potential fatal errors.